internal/mtls/mtls.go

```diff
@@ -0,0 +1,130 @@
+		zap.String("secretNamespace", secretNamespace))
+
+	secret, err := clientset.CoreV1().Secrets(secretNamespace).Get(ctx, secretName, metav1.GetOptions{})
+	if err != nil {
+		logger.Error("Failed to get mTLS secret from Kubernetes API", zap.Error(err))
+		return nil, nil, nil, fmt.Errorf("failed to get secret '%s' in namespace '%s': %w", secretName, secretNamespace, err)
+	}
+
+	caData, ok := secret.Data["ca.crt"]
+	if !ok || len(caData) == 0 {
+		return nil, nil, nil, fmt.Errorf("%w: 'ca.crt' key", ErrCertDataMissingInSecret)
+	}
+	logger.Debug("Successfully retrieved ca.crt from secret.")
+
+	clientCertData, ok = secret.Data["tls.crt"]
+	if !ok || len(clientCertData) == 0 {
+		return nil, nil, nil, fmt.Errorf("%w: 'tls.crt' key (client certificate)", ErrCertDataMissingInSecret)
+	}
+	logger.Debug("Successfully retrieved tls.crt (client certificate) from secret.")
+
+	clientKeyData, ok = secret.Data["tls.key"]
+	if !ok || len(clientKeyData) == 0 {
+		return nil, nil, nil, fmt.Errorf("%w: 'tls.key' key (client key)", ErrCertDataMissingInSecret)
+	}
+	logger.Debug("Successfully retrieved tls.key (client key) from secret.")
+
+	return caData, clientCertData, clientKeyData, nil
+}
+
+// loadMTLSCredentialsFromData loads mTLS credentials from byte slice.
+func loadMTLSCredentialsFromData(logger *zap.Logger, aksManaged bool, caCertData, clientCertData, clientKeyData []byte) (credentials.TransportCredentials, error) {
+	// Validate PEM data
+	if !isValidPEM(caCertData) || !isValidPEM(clientCertData) || !isValidPEM(clientKeyData) {
+		return nil, errors.New("invalid PEM data: ensure proper headers and formatting")
+	}
+
+	// Load client certificate and key from byte slices
+	cert, err := tls.X509KeyPair(clientCertData, clientKeyData)
+	if err != nil {
+		return nil, fmt.Errorf("failed to load client certificate and key from data: %w", err)
+	}
+	logger.Debug("Client certificate and key loaded from data.")
+
+	// Load CA certificate from byte slice
+	certPool := x509.NewCertPool()
+	if !certPool.AppendCertsFromPEM(caCertData) {
+		return nil, errors.New("failed to append CA certificate to pool from data")
+	}
+	logger.Debug("CA certificate appended to pool from data.")
+
+	tlsConfig := &tls.Config{
+		ServerName:         hubbleRelayExpectedServerName,
+		Certificates:       []tls.Certificate{cert},
+		RootCAs:            certPool,
+		InsecureSkipVerify: false,
+		MinVersion:         tls.VersionTLS12,
+	}
+
+	var cred credentials.TransportCredentials
+	if aksManaged {
+		logger.Debug("AKS Managed: Using experimental NewTLSWithALPNDisabled")
+		cred = exp_credentials.NewTLSWithALPNDisabled(tlsConfig)
+	}
+
+	return cred, nil
+}
+
+// FetchAndLoadMTLSCredentials encapsulates the logic for fetching and loading mTLS credentials.
+func FetchAndLoadMTLSCredentials(ctx context.Context, clientset kubernetes.Interface, logger *zap.Logger, secretName, namespace string, isAKSManaged bool) (credentials.TransportCredentials, error) {
```

Comment:
1. What's the difference between "fetching" and "loading"?
1. Functions "encapsulate" code by definition. No need to comment about that.
1. This logic is very specific to how Hubble shares client credentials. Rename to reflect that.
```suggestion
// GetHubbleTransportCredentials read TLS transport credentials to connect to Hubble from the given Secret.
func GetHubbleCTransportCredentials(ctx context.Context, clientset kubernetes.Interface, logger *zap.Logger, secretName, namespace string, isAKSManaged bool) (credentials.TransportCredentials, error) {
```